lib/publiccloud/basetest.pm

```diff
@@ -103,36 +103,56 @@ sub provider_factory {
 
 sub cleanup {
     # to be overridden by tests
+    # non-destructive cleanup for test module started by post_run_hook
```

Comment:
I'm very much confused by this. What is the purpose of a non-destructive cleanup after each test module? Where is this or will this used?